Microsoft.Toolkit.Uwp.UI.Controls.DataGrid/DataGrid/DataGrid.xaml

```diff
@@ -23,8 +23,8 @@
             <SolidColorBrush x:Key="FillerGridLinesBrush" Color="Transparent"/>
             <StaticResource x:Key="ScrollBarsSeparatorBackground" ResourceKey="SystemControlPageBackgroundChromeLowBrush"/>
             <StaticResource x:Key="DataGridColumnHeaderForegroundBrush" ResourceKey="SystemControlForegroundBaseMediumBrush"/>
-            <StaticResource x:Key="DataGridColumnHeaderBackgroundColor" ResourceKey="SystemAltHighColor"/>
-            <SolidColorBrush x:Key="DataGridColumnHeaderBackgroundBrush" Color="{StaticResource DataGridColumnHeaderBackgroundColor}"/>
+            <Color x:Key="DataGridColumnHeaderBackgroundColor">#FF000000</Color>
```

Comment:
You should keep using the system color resources to be sure that everything will render properly in dark/light theme but also all high contrast modes

Reply:
The HEX values are the equivalent for "SystemAltHighColor" (for high contrast, light and dark theme). The DataGrid is already using HEX colors in other places. However if system color resources is a must in this case, would need to generate the Color in the code behind based on SystemAltHighColor. Any ideas would be appreciated.

Reply:
Maybe you can try to remove the `DataGridColumnHeaderBackgroundColor` and consume directly the `SystemAltHighColor` as a *theme resource* in `DataGridColumnHeaderBackgroundBrush` ?

```xaml
<SolidColorBrush x:Key="DataGridColumnHeaderBackgroundBrush" Color="{ThemeResource SystemAltHighColor}"/>
```

The `<StaticResource />`  redirection might be the one preventing you to easily redefine the color. Removing the color intermediate step can simplify the overloading.

Reply:
Hi @vgromfeld, as you may see in the issue description, SolidColorBrush is the cause of the bug, is not supported by the ColorAnimation and as a result you can't change the color. I found that  <Color/> works and allows to customize the color later.  If you use SolidColorBrush directly or a StaticResource with SolidColorBrush (the original code) is not possible to customize it. I'm still investigating how to use the Colors without the HEX to future proof the code in case Windows changes the values of SystemAltHight.

Reply:
Yeah, we definitely don't want to be hard-coding values here. @RBrid any suggestions?

Reply:
I've tried the following code in XAML studio and it is working fine.
There is something 

```xaml
<UserControl>
<StackPanel>
    <StackPanel.Resources>
        <ResourceDictionary>
            <ResourceDictionary.ThemeDictionaries>
                <ResourceDictionary x:Key="Dark">
                    <StaticResource x:Key="DataGridColumnHeaderBackgroundColor" ResourceKey="SystemAltHighColor"/>
                    <StaticResource x:Key="DataGridColumnHeaderBackgroundColor2" ResourceKey="SystemAccentColor"/>
                    <SolidColorBrush x:Key="DataGridColumnHeaderBackgroundBrush" Color="{StaticResource DataGridColumnHeaderBackgroundColor}"/>
                </ResourceDictionary>
                <ResourceDictionary x:Key="Light">
                    <StaticResource x:Key="DataGridColumnHeaderBackgroundColor" ResourceKey="SystemAltHighColor"/>
                    <StaticResource x:Key="DataGridColumnHeaderBackgroundColor2" ResourceKey="SystemAccentColor"/>
                    <SolidColorBrush x:Key="DataGridColumnHeaderBackgroundBrush" Color="{StaticResource DataGridColumnHeaderBackgroundColor}"/>
                </ResourceDictionary>
                <ResourceDictionary x:Key="HighContrast">
                    <StaticResource x:Key="DataGridColumnHeaderBackgroundColor" ResourceKey="SystemAltHighColor"/>
                    <StaticResource x:Key="DataGridColumnHeaderBackgroundColor2" ResourceKey="SystemAccentColor"/>
                    <SolidColorBrush x:Key="DataGridColumnHeaderBackgroundBrush" Color="{StaticResource DataGridColumnHeaderBackgroundColor}"/>
                </ResourceDictionary>
            </ResourceDictionary.ThemeDictionaries>             
        </ResourceDictionary>
    </StackPanel.Resources>
    <VisualStateManager.VisualStateGroups>
        <VisualStateGroup>
            <VisualState>
                <VisualState.StateTriggers>
                    <AdaptiveTrigger MinWindowWidth="100" />
                </VisualState.StateTriggers>
                <Storyboard>
                    <ColorAnimation 
                    Storyboard.TargetName="BackgroundRectangle"
                    Storyboard.TargetProperty="(Border.Background).(SolidColorBrush.Color)"
                    Duration="0"
                    To="{ThemeResource DataGridColumnHeaderBackgroundColor2}"/>
                </Storyboard>
            </VisualState>
        </VisualStateGroup>
    </VisualStateManager.VisualStateGroups>
    <Border x:Name="BackgroundRectangle" 
        Width="100"
        Height="100" 
        Background="{ThemeResource DataGridColumnHeaderBackgroundBrush}" />
</StackPanel> 
</UserControl>
```

It is still unclear to me why this is not working for the `DataGrid`.

Reply:
@forerunnerG34 can you test out @vgromfeld's suggestion here in the resource dictionary in the toolkit? If it's still not working, we'll have to follow-up with @RBrid.

FYI @anawishnoff